src/Instrument/FileSystem/Enumerator.php

```diff
@@ -54,23 +63,71 @@ public function __construct($rootDirectory, array $includePaths = [], array $exc
     /**
      * Returns an enumerator for files
      *
-     * @return \CallbackFilterIterator|\RecursiveIteratorIterator|\SplFileInfo[]
+     * @return \Iterator|SplFileInfo[]
+     * @throws UnexpectedValueException
+     * @throws InvalidArgumentException
+     * @throws LogicException
      */
     public function enumerate()
     {
-        $iterator = new \RecursiveIteratorIterator(
-            new \RecursiveDirectoryIterator(
-                $this->rootDirectory,
-                \FilesystemIterator::SKIP_DOTS | \FilesystemIterator::UNIX_PATHS
-            )
-        );
+        $finder = new Finder();
```

Comment:
I would recommend using `Finder` as a dependency for now. Should make up for a more solid test.

Also I just realized, that this operation here is a bit IO heavy.
If you call `enumerate` a number of times you will always hit IO to scan again for files.
I get that there could be new files added between two concurrent runs, but I guess this would not happen on a production environment.

Maybe @lisachenko could say more about this ;)

Reply:
Well, the IO heavy operation is there anyway, whether with Finder or using the old way... I didn't want to mess with the original design, this was supposed to be just an optimization... hopefully there won't be  a problem with the new dependency, since it isn't dependent of anything else...

Reply:
@icanhazstring not sure what you meant with the requested changes - just the typo or also something regarding this conversation?

Reply:
Ye I know that this was like this before. Just realized it by now.
And after thinking this has nothing to do with this PR - might be subject for another PR - so for me everything is fine 👍 

@lisachenko just has to accept/merge it ;)